src/controllers/RateController.ts

```diff
@@ -1,16 +1,18 @@
 import { Request, Response } from 'express';
 
 import { RateSourceService } from '../services/rateSource/RateSourceService';
+import { CurrencyExchangeRateService } from '../services/CurrencyExchangeRateService';
 
 class RateController {
-  constructor(private readonly rateSourceService: RateSourceService) {}
+  constructor(private readonly rateSourceService?: RateSourceService) {}
 
   public async getRate(
     _: Request,
     res: Response,
   ): Promise<Response | undefined> {
-    const { rate, code, errorMessage } =
-      await this.rateSourceService.retrieve();
+    const { rate, code, errorMessage } = await new CurrencyExchangeRateService(
```

Comment:
What about creating an instance of CurrencyExchangeRateService in the constructor?